### ISSUE TITLE
nan as input value returns optimal f-value on Rastrigin function

### ISSUE THREAD

**Issue by nikohansen** [numbbo/coco#769]
```python
>>> from numpy import nan
>>> import cocoex as ex
>>> suite = ex.Suite("bbob", "", "")
>>> p = suite.get_problem(suite.ids('f015', 'd02')[0])
>>> print(p([1, 1, ]))
1028.2944387
>>> print(p([nan, nan, ]))
1000.0
```

**Comment by nikohansen** [comment]
more generally, `nan` leads to finite return values on a bunch of functions:
```python

from numpy import nan
import cocoex as ex
suite = ex.Suite("bbob", "", "")
for p in suite:
    if isfinite(p(p.dimension * [nan])):
        print(p.id, p(p.dimension * [nan]))
        
bbob_f002_i01_d02 -209.88
bbob_f002_i02_d02 -92.09
bbob_f002_i03_d02 -87.89
bbob_f002_i04_d02 320.19
bbob_f002_i05_d02 -104.24
bbob_f002_i51_d02 -184.31
bbob_f002_i52_d02 56.84
bbob_f002_i53_d02 -189.73
bbob_f002_i54_d02 465.97
bbob_f002_i55_d02 25.74
bbob_f002_i56_d02 41.67
bbob_f002_i57_d02 -190.95
bbob_f002_i58_d02 1.37
bbob_f002_i59_d02 -164.58
bbob_f002_i60_d02 -180.38
bbob_f003_i01_d02 -462.09
bbob_f003_i02_d02 77.66
bbob_f003_i03_d02 115.68
bbob_f003_i04_d02 -3.71
bbob_f003_i05_d02 132.18
bbob_f003_i51_d02 64.38
bbob_f003_i52_d02 282.48
bbob_f003_i53_d02 482.61
bbob_f003_i54_d02 71.77
bbob_f003_i55_d02 844.88
bbob_f003_i56_d02 -42.22
bbob_f003_i57_d02 -91.79
bbob_f003_i58_d02 -12.77
bbob_f003_i59_d02 -1000.0
bbob_f003_i60_d02 -1000.0
bbob_f004_i01_d02 -462.09
bbob_f004_i02_d02 77.66
bbob_f004_i03_d02 115.68
bbob_f004_i04_d02 -3.71
bbob_f004_i05_d02 132.18
bbob_f004_i51_d02 64.38
bbob_f004_i52_d02 282.48
bbob_f004_i53_d02 482.61
bbob_f004_i54_d02 71.77
bbob_f004_i55_d02 844.88
bbob_f004_i56_d02 -42.22
bbob_f004_i57_d02 -91.79
bbob_f004_i58_d02 -12.77
bbob_f004_i59_d02 -1000.0
bbob_f004_i60_d02 -1000.0
bbob_f010_i01_d02 -54.94
bbob_f010_i02_d02 59.13
bbob_f010_i03_d02 -491.53
bbob_f010_i04_d02 -170.2
bbob_f010_i05_d02 -469.75
bbob_f010_i51_d02 183.27
bbob_f010_i52_d02 -1000.0
bbob_f010_i53_d02 -56.58
bbob_f010_i54_d02 -204.37
bbob_f010_i55_d02 8.55
bbob_f010_i56_d02 325.87
bbob_f010_i57_d02 -95.25
bbob_f010_i58_d02 49.43
bbob_f010_i59_d02 -97.72
bbob_f010_i60_d02 -51.0
bbob_f011_i01_d02 76.27
bbob_f011_i02_d02 -22.55
bbob_f011_i03_d02 13.66
bbob_f011_i04_d02 937.84
bbob_f011_i05_d02 -78.72
bbob_f011_i51_d02 -69.49
bbob_f011_i52_d02 -27.18
bbob_f011_i53_d02 -696.88
bbob_f011_i54_d02 -489.53
bbob_f011_i55_d02 -40.04
bbob_f011_i56_d02 -103.81
bbob_f011_i57_d02 -23.8
bbob_f011_i58_d02 74.59
bbob_f011_i59_d02 222.59
bbob_f011_i60_d02 -297.77
bbob_f015_i01_d02 1000.0
bbob_f015_i02_d02 70.03
bbob_f015_i03_d02 -48.22
bbob_f015_i04_d02 25.47
bbob_f015_i05_d02 -100.81
bbob_f015_i51_d02 -34.96
bbob_f015_i52_d02 -272.79
bbob_f015_i53_d02 141.02
bbob_f015_i54_d02 81.13
bbob_f015_i55_d02 104.92
bbob_f015_i56_d02 30.29
bbob_f015_i57_d02 121.05
bbob_f015_i58_d02 -10.33
bbob_f015_i59_d02 -68.46
bbob_f015_i60_d02 -131.49
bbob_f016_i01_d02 71.35
bbob_f016_i02_d02 -355.22
bbob_f016_i03_d02 99.17
bbob_f016_i04_d02 -517.91
bbob_f016_i05_d02 -57.21
bbob_f016_i51_d02 -1000.0
bbob_f016_i52_d02 -65.54
bbob_f016_i53_d02 -23.75
bbob_f016_i54_d02 35.13
bbob_f016_i55_d02 -79.82
bbob_f016_i56_d02 192.58
bbob_f016_i57_d02 -268.48
bbob_f016_i58_d02 -1000.0
bbob_f016_i59_d02 -13.88
bbob_f016_i60_d02 -124.04
bbob_f002_i01_d03 -209.88
bbob_f002_i02_d03 -92.09
bbob_f002_i03_d03 -87.89
bbob_f002_i04_d03 320.19
bbob_f002_i05_d03 -104.24
bbob_f002_i51_d03 -184.31
bbob_f002_i52_d03 56.84
bbob_f002_i53_d03 -189.73
bbob_f002_i54_d03 465.97
bbob_f002_i55_d03 25.74
bbob_f002_i56_d03 41.67
bbob_f002_i57_d03 -190.95
bbob_f002_i58_d03 1.37
bbob_f002_i59_d03 -164.58
bbob_f002_i60_d03 -180.38
bbob_f003_i01_d03 -462.09
bbob_f003_i02_d03 77.66
bbob_f003_i03_d03 115.68
bbob_f003_i04_d03 -3.71
bbob_f003_i05_d03 132.18
bbob_f003_i51_d03 64.38
bbob_f003_i52_d03 282.48
bbob_f003_i53_d03 482.61
bbob_f003_i54_d03 71.77
bbob_f003_i55_d03 844.88
bbob_f003_i56_d03 -42.22
bbob_f003_i57_d03 -91.79
bbob_f003_i58_d03 -12.77
bbob_f003_i59_d03 -1000.0
bbob_f003_i60_d03 -1000.0
bbob_f004_i01_d03 -462.09
bbob_f004_i02_d03 77.66
bbob_f004_i03_d03 115.68
bbob_f004_i04_d03 -3.71
bbob_f004_i05_d03 132.18
bbob_f004_i51_d03 64.38
bbob_f004_i52_d03 282.48
bbob_f004_i53_d03 482.61
bbob_f004_i54_d03 71.77
bbob_f004_i55_d03 844.88
bbob_f004_i56_d03 -42.22
bbob_f004_i57_d03 -91.79
bbob_f004_i58_d03 -12.77
bbob_f004_i59_d03 -1000.0
bbob_f004_i60_d03 -1000.0
bbob_f010_i01_d03 -54.94
bbob_f010_i02_d03 59.13
bbob_f010_i03_d03 -491.53
bbob_f010_i04_d03 -170.2
bbob_f010_i05_d03 -469.75
bbob_f010_i51_d03 183.27
bbob_f010_i52_d03 -1000.0
bbob_f010_i53_d03 -56.58
bbob_f010_i54_d03 -204.37
bbob_f010_i55_d03 8.55
bbob_f010_i56_d03 325.87
bbob_f010_i57_d03 -95.25
bbob_f010_i58_d03 49.43
bbob_f010_i59_d03 -97.72
bbob_f010_i60_d03 -51.0
bbob_f011_i01_d03 76.27
bbob_f011_i02_d03 -22.55
bbob_f011_i03_d03 13.66
bbob_f011_i04_d03 937.84
bbob_f011_i05_d03 -78.72
bbob_f011_i51_d03 -69.49
bbob_f011_i52_d03 -27.18
bbob_f011_i53_d03 -696.88
bbob_f011_i54_d03 -489.53
bbob_f011_i55_d03 -40.04
bbob_f011_i56_d03 -103.81
bbob_f011_i57_d03 -23.8
bbob_f011_i58_d03 74.59
bbob_f011_i59_d03 222.59
bbob_f011_i60_d03 -297.77
bbob_f015_i01_d03 1000.0
bbob_f015_i02_d03 70.03
bbob_f015_i03_d03 -48.22
bbob_f015_i04_d03 25.47
bbob_f015_i05_d03 -100.81
bbob_f015_i51_d03 -34.96
bbob_f015_i52_d03 -272.79
bbob_f015_i53_d03 141.02
bbob_f015_i54_d03 81.13
bbob_f015_i55_d03 104.92
bbob_f015_i56_d03 30.29
bbob_f015_i57_d03 121.05
bbob_f015_i58_d03 -10.33
bbob_f015_i59_d03 -68.46
bbob_f015_i60_d03 -131.49
bbob_f016_i01_d03 71.35
bbob_f016_i02_d03 -355.22
bbob_f016_i03_d03 99.17
bbob_f016_i04_d03 -517.91
bbob_f016_i05_d03 -57.21
bbob_f016_i51_d03 -1000.0
bbob_f016_i52_d03 -65.54
bbob_f016_i53_d03 -23.75
bbob_f016_i54_d03 35.13
bbob_f016_i55_d03 -79.82
bbob_f016_i56_d03 192.58
bbob_f016_i57_d03 -268.48
bbob_f016_i58_d03 -1000.0
bbob_f016_i59_d03 -13.88
bbob_f016_i60_d03 -124.04
bbob_f002_i01_d05 -209.88
bbob_f002_i02_d05 -92.09
bbob_f002_i03_d05 -87.89
bbob_f002_i04_d05 320.19
bbob_f002_i05_d05 -104.24
bbob_f002_i51_d05 -184.31
bbob_f002_i52_d05 56.84
bbob_f002_i53_d05 -189.73
bbob_f002_i54_d05 465.97
bbob_f002_i55_d05 25.74
bbob_f002_i56_d05 41.67
bbob_f002_i57_d05 -190.95
bbob_f002_i58_d05 1.37
bbob_f002_i59_d05 -164.58
bbob_f002_i60_d05 -180.38
bbob_f003_i01_d05 -462.09
bbob_f003_i02_d05 77.66
bbob_f003_i03_d05 115.68
bbob_f003_i04_d05 -3.71
bbob_f003_i05_d05 132.18
bbob_f003_i51_d05 64.38
bbob_f003_i52_d05 282.48
bbob_f003_i53_d05 482.61
bbob_f003_i54_d05 71.77
bbob_f003_i55_d05 844.88
bbob_f003_i56_d05 -42.22
bbob_f003_i57_d05 -91.79
bbob_f003_i58_d05 -12.77
bbob_f003_i59_d05 -1000.0
bbob_f003_i60_d05 -1000.0
bbob_f004_i01_d05 -462.09
bbob_f004_i02_d05 77.66
bbob_f004_i03_d05 115.68
bbob_f004_i04_d05 -3.71
bbob_f004_i05_d05 132.18
bbob_f004_i51_d05 64.38
bbob_f004_i52_d05 282.48
bbob_f004_i53_d05 482.61
bbob_f004_i54_d05 71.77
bbob_f004_i55_d05 844.88
bbob_f004_i56_d05 -42.22
bbob_f004_i57_d05 -91.79
bbob_f004_i58_d05 -12.77
bbob_f004_i59_d05 -1000.0
bbob_f004_i60_d05 -1000.0
bbob_f010_i01_d05 -54.94
bbob_f010_i02_d05 59.13
bbob_f010_i03_d05 -491.53
bbob_f010_i04_d05 -170.2
bbob_f010_i05_d05 -469.75
bbob_f010_i51_d05 183.27
bbob_f010_i52_d05 -1000.0
bbob_f010_i53_d05 -56.58
bbob_f010_i54_d05 -204.37
bbob_f010_i55_d05 8.55
bbob_f010_i56_d05 325.87
bbob_f010_i57_d05 -95.25
bbob_f010_i58_d05 49.43
bbob_f010_i59_d05 -97.72
bbob_f010_i60_d05 -51.0
bbob_f011_i01_d05 76.27
bbob_f011_i02_d05 -22.55
bbob_f011_i03_d05 13.66
bbob_f011_i04_d05 937.84
bbob_f011_i05_d05 -78.72
bbob_f011_i51_d05 -69.49
bbob_f011_i52_d05 -27.18
bbob_f011_i53_d05 -696.88
bbob_f011_i54_d05 -489.53
bbob_f011_i55_d05 -40.04
bbob_f011_i56_d05 -103.81
bbob_f011_i57_d05 -23.8
bbob_f011_i58_d05 74.59
bbob_f011_i59_d05 222.59
bbob_f011_i60_d05 -297.77
bbob_f015_i01_d05 1000.0
bbob_f015_i02_d05 70.03
bbob_f015_i03_d05 -48.22
bbob_f015_i04_d05 25.47
bbob_f015_i05_d05 -100.81
bbob_f015_i51_d05 -34.96
bbob_f015_i52_d05 -272.79
bbob_f015_i53_d05 141.02
bbob_f015_i54_d05 81.13
bbob_f015_i55_d05 104.92
bbob_f015_i56_d05 30.29
bbob_f015_i57_d05 121.05
bbob_f015_i58_d05 -10.33
bbob_f015_i59_d05 -68.46
bbob_f015_i60_d05 -131.49
bbob_f016_i01_d05 71.35
bbob_f016_i02_d05 -355.22
bbob_f016_i03_d05 99.17
bbob_f016_i04_d05 -517.91
bbob_f016_i05_d05 -57.21
bbob_f016_i51_d05 -1000.0
bbob_f016_i52_d05 -65.54
bbob_f016_i53_d05 -23.75
bbob_f016_i54_d05 35.13
bbob_f016_i55_d05 -79.82
bbob_f016_i56_d05 192.58
bbob_f016_i57_d05 -268.48
bbob_f016_i58_d05 -1000.0
bbob_f016_i59_d05 -13.88
bbob_f016_i60_d05 -124.04
bbob_f002_i01_d10 -209.88
bbob_f002_i02_d10 -92.09
bbob_f002_i03_d10 -87.89
bbob_f002_i04_d10 320.19
bbob_f002_i05_d10 -104.24
bbob_f002_i51_d10 -184.31
bbob_f002_i52_d10 56.84
bbob_f002_i53_d10 -189.73
bbob_f002_i54_d10 465.97
bbob_f002_i55_d10 25.74
bbob_f002_i56_d10 41.67
bbob_f002_i57_d10 -190.95
bbob_f002_i58_d10 1.37
bbob_f002_i59_d10 -164.58
bbob_f002_i60_d10 -180.38
bbob_f003_i01_d10 -462.09
bbob_f003_i02_d10 77.66
bbob_f003_i03_d10 115.68
bbob_f003_i04_d10 -3.71
bbob_f003_i05_d10 132.18
bbob_f003_i51_d10 64.38
bbob_f003_i52_d10 282.48
bbob_f003_i53_d10 482.61
bbob_f003_i54_d10 71.77
bbob_f003_i55_d10 844.88
bbob_f003_i56_d10 -42.22
bbob_f003_i57_d10 -91.79
bbob_f003_i58_d10 -12.77
bbob_f003_i59_d10 -1000.0
bbob_f003_i60_d10 -1000.0
bbob_f004_i01_d10 -462.09
bbob_f004_i02_d10 77.66
bbob_f004_i03_d10 115.68
bbob_f004_i04_d10 -3.71
bbob_f004_i05_d10 132.18
bbob_f004_i51_d10 64.38
bbob_f004_i52_d10 282.48
bbob_f004_i53_d10 482.61
bbob_f004_i54_d10 71.77
bbob_f004_i55_d10 844.88
bbob_f004_i56_d10 -42.22
bbob_f004_i57_d10 -91.79
bbob_f004_i58_d10 -12.77
bbob_f004_i59_d10 -1000.0
bbob_f004_i60_d10 -1000.0
bbob_f010_i01_d10 -54.94
bbob_f010_i02_d10 59.13
bbob_f010_i03_d10 -491.53
bbob_f010_i04_d10 -170.2
bbob_f010_i05_d10 -469.75
bbob_f010_i51_d10 183.27
bbob_f010_i52_d10 -1000.0
bbob_f010_i53_d10 -56.58
bbob_f010_i54_d10 -204.37
bbob_f010_i55_d10 8.55
bbob_f010_i56_d10 325.87
bbob_f010_i57_d10 -95.25
bbob_f010_i58_d10 49.43
bbob_f010_i59_d10 -97.72
bbob_f010_i60_d10 -51.0
bbob_f011_i01_d10 76.27
bbob_f011_i02_d10 -22.55
bbob_f011_i03_d10 13.66
bbob_f011_i04_d10 937.84
bbob_f011_i05_d10 -78.72
bbob_f011_i51_d10 -69.49
bbob_f011_i52_d10 -27.18
bbob_f011_i53_d10 -696.88
bbob_f011_i54_d10 -489.53
bbob_f011_i55_d10 -40.04
bbob_f011_i56_d10 -103.81
bbob_f011_i57_d10 -23.8
bbob_f011_i58_d10 74.59
bbob_f011_i59_d10 222.59
bbob_f011_i60_d10 -297.77
bbob_f015_i01_d10 1000.0
bbob_f015_i02_d10 70.03
bbob_f015_i03_d10 -48.22
bbob_f015_i04_d10 25.47
bbob_f015_i05_d10 -100.81
bbob_f015_i51_d10 -34.96
bbob_f015_i52_d10 -272.79
bbob_f015_i53_d10 141.02
bbob_f015_i54_d10 81.13
bbob_f015_i55_d10 104.92
bbob_f015_i56_d10 30.29
bbob_f015_i57_d10 121.05
bbob_f015_i58_d10 -10.33
bbob_f015_i59_d10 -68.46
bbob_f015_i60_d10 -131.49
bbob_f016_i01_d10 71.35
bbob_f016_i02_d10 -355.22
bbob_f016_i03_d10 99.17
bbob_f016_i04_d10 -517.91
bbob_f016_i05_d10 -57.21
bbob_f016_i51_d10 -1000.0
bbob_f016_i52_d10 -65.54
bbob_f016_i53_d10 -23.75
bbob_f016_i54_d10 35.13
bbob_f016_i55_d10 -79.82
bbob_f016_i56_d10 192.58
bbob_f016_i57_d10 -268.48
bbob_f016_i58_d10 -1000.0
bbob_f016_i59_d10 -13.88
bbob_f016_i60_d10 -124.04
bbob_f002_i01_d20 -209.88
bbob_f002_i02_d20 -92.09
bbob_f002_i03_d20 -87.89
bbob_f002_i04_d20 320.19
bbob_f002_i05_d20 -104.24
bbob_f002_i51_d20 -184.31
bbob_f002_i52_d20 56.84
bbob_f002_i53_d20 -189.73
bbob_f002_i54_d20 465.97
bbob_f002_i55_d20 25.74
bbob_f002_i56_d20 41.67
bbob_f002_i57_d20 -190.95
bbob_f002_i58_d20 1.37
bbob_f002_i59_d20 -164.58
bbob_f002_i60_d20 -180.38
bbob_f003_i01_d20 -462.09
bbob_f003_i02_d20 77.66
bbob_f003_i03_d20 115.68
bbob_f003_i04_d20 -3.71
bbob_f003_i05_d20 132.18
bbob_f003_i51_d20 64.38
bbob_f003_i52_d20 282.48
bbob_f003_i53_d20 482.61
bbob_f003_i54_d20 71.77
bbob_f003_i55_d20 844.88
bbob_f003_i56_d20 -42.22
bbob_f003_i57_d20 -91.79
bbob_f003_i58_d20 -12.77
bbob_f003_i59_d20 -1000.0
bbob_f003_i60_d20 -1000.0
bbob_f004_i01_d20 -462.09
bbob_f004_i02_d20 77.66
bbob_f004_i03_d20 115.68
bbob_f004_i04_d20 -3.71
bbob_f004_i05_d20 132.18
bbob_f004_i51_d20 64.38
bbob_f004_i52_d20 282.48
bbob_f004_i53_d20 482.61
bbob_f004_i54_d20 71.77
bbob_f004_i55_d20 844.88
bbob_f004_i56_d20 -42.22
bbob_f004_i57_d20 -91.79
bbob_f004_i58_d20 -12.77
bbob_f004_i59_d20 -1000.0
bbob_f004_i60_d20 -1000.0
bbob_f010_i01_d20 -54.94
bbob_f010_i02_d20 59.13
bbob_f010_i03_d20 -491.53
bbob_f010_i04_d20 -170.2
bbob_f010_i05_d20 -469.75
bbob_f010_i51_d20 183.27
bbob_f010_i52_d20 -1000.0
bbob_f010_i53_d20 -56.58
bbob_f010_i54_d20 -204.37
bbob_f010_i55_d20 8.55
bbob_f010_i56_d20 325.87
bbob_f010_i57_d20 -95.25
bbob_f010_i58_d20 49.43
bbob_f010_i59_d20 -97.72
bbob_f010_i60_d20 -51.0
bbob_f011_i01_d20 76.27
bbob_f011_i02_d20 -22.55
bbob_f011_i03_d20 13.66
bbob_f011_i04_d20 937.84
bbob_f011_i05_d20 -78.72
bbob_f011_i51_d20 -69.49
bbob_f011_i52_d20 -27.18
bbob_f011_i53_d20 -696.88
bbob_f011_i54_d20 -489.53
bbob_f011_i55_d20 -40.04
bbob_f011_i56_d20 -103.81
bbob_f011_i57_d20 -23.8
bbob_f011_i58_d20 74.59
bbob_f011_i59_d20 222.59
bbob_f011_i60_d20 -297.77
bbob_f015_i01_d20 1000.0
bbob_f015_i02_d20 70.03
bbob_f015_i03_d20 -48.22
bbob_f015_i04_d20 25.47
bbob_f015_i05_d20 -100.81
bbob_f015_i51_d20 -34.96
bbob_f015_i52_d20 -272.79
bbob_f015_i53_d20 141.02
bbob_f015_i54_d20 81.13
bbob_f015_i55_d20 104.92
bbob_f015_i56_d20 30.29
bbob_f015_i57_d20 121.05
bbob_f015_i58_d20 -10.33
bbob_f015_i59_d20 -68.46
bbob_f015_i60_d20 -131.49
bbob_f016_i01_d20 71.35
bbob_f016_i02_d20 -355.22
bbob_f016_i03_d20 99.17
bbob_f016_i04_d20 -517.91
bbob_f016_i05_d20 -57.21
bbob_f016_i51_d20 -1000.0
bbob_f016_i52_d20 -65.54
bbob_f016_i53_d20 -23.75
bbob_f016_i54_d20 35.13
bbob_f016_i55_d20 -79.82
bbob_f016_i56_d20 192.58
bbob_f016_i57_d20 -268.48
bbob_f016_i58_d20 -1000.0
bbob_f016_i59_d20 -13.88
bbob_f016_i60_d20 -124.04
bbob_f002_i01_d40 -209.88
bbob_f002_i02_d40 -92.09
bbob_f002_i03_d40 -87.89
bbob_f002_i04_d40 320.19
bbob_f002_i05_d40 -104.24
bbob_f002_i51_d40 -184.31
bbob_f002_i52_d40 56.84
bbob_f002_i53_d40 -189.73
bbob_f002_i54_d40 465.97
bbob_f002_i55_d40 25.74
bbob_f002_i56_d40 41.67
bbob_f002_i57_d40 -190.95
bbob_f002_i58_d40 1.37
bbob_f002_i59_d40 -164.58
bbob_f002_i60_d40 -180.38
bbob_f003_i01_d40 -462.09
bbob_f003_i02_d40 77.66
bbob_f003_i03_d40 115.68
bbob_f003_i04_d40 -3.71
bbob_f003_i05_d40 132.18
bbob_f003_i51_d40 64.38
bbob_f003_i52_d40 282.48
bbob_f003_i53_d40 482.61
bbob_f003_i54_d40 71.77
bbob_f003_i55_d40 844.88
bbob_f003_i56_d40 -42.22
bbob_f003_i57_d40 -91.79
bbob_f003_i58_d40 -12.77
bbob_f003_i59_d40 -1000.0
bbob_f003_i60_d40 -1000.0
bbob_f004_i01_d40 -462.09
bbob_f004_i02_d40 77.66
bbob_f004_i03_d40 115.68
bbob_f004_i04_d40 -3.71
bbob_f004_i05_d40 132.18
bbob_f004_i51_d40 64.38
bbob_f004_i52_d40 282.48
bbob_f004_i53_d40 482.61
bbob_f004_i54_d40 71.77
bbob_f004_i55_d40 844.88
bbob_f004_i56_d40 -42.22
bbob_f004_i57_d40 -91.79
bbob_f004_i58_d40 -12.77
bbob_f004_i59_d40 -1000.0
bbob_f004_i60_d40 -1000.0
bbob_f010_i01_d40 -54.94
bbob_f010_i02_d40 59.13
bbob_f010_i03_d40 -491.53
bbob_f010_i04_d40 -170.2
bbob_f010_i05_d40 -469.75
bbob_f010_i51_d40 183.27
bbob_f010_i52_d40 -1000.0
bbob_f010_i53_d40 -56.58
bbob_f010_i54_d40 -204.37
bbob_f010_i55_d40 8.55
bbob_f010_i56_d40 325.87
bbob_f010_i57_d40 -95.25
bbob_f010_i58_d40 49.43
bbob_f010_i59_d40 -97.72
bbob_f010_i60_d40 -51.0
bbob_f011_i01_d40 76.27
bbob_f011_i02_d40 -22.55
bbob_f011_i03_d40 13.66
bbob_f011_i04_d40 937.84
bbob_f011_i05_d40 -78.72
bbob_f011_i51_d40 -69.49
bbob_f011_i52_d40 -27.18
bbob_f011_i53_d40 -696.88
bbob_f011_i54_d40 -489.53
bbob_f011_i55_d40 -40.04
bbob_f011_i56_d40 -103.81
bbob_f011_i57_d40 -23.8
bbob_f011_i58_d40 74.59
bbob_f011_i59_d40 222.59
bbob_f011_i60_d40 -297.77
bbob_f015_i01_d40 1000.0
bbob_f015_i02_d40 70.03
bbob_f015_i03_d40 -48.22
bbob_f015_i04_d40 25.47
bbob_f015_i05_d40 -100.81
bbob_f015_i51_d40 -34.96
bbob_f015_i52_d40 -272.79
bbob_f015_i53_d40 141.02
bbob_f015_i54_d40 81.13
bbob_f015_i55_d40 104.92
bbob_f015_i56_d40 30.29
bbob_f015_i57_d40 121.05
bbob_f015_i58_d40 -10.33
bbob_f015_i59_d40 -68.46
bbob_f015_i60_d40 -131.49
bbob_f016_i01_d40 71.35
bbob_f016_i02_d40 -355.22
bbob_f016_i03_d40 99.17
bbob_f016_i04_d40 -517.91
bbob_f016_i05_d40 -57.21
bbob_f016_i51_d40 -1000.0
bbob_f016_i52_d40 -65.54
bbob_f016_i53_d40 -23.75
bbob_f016_i54_d40 35.13
bbob_f016_i55_d40 -79.82
bbob_f016_i56_d40 192.58
bbob_f016_i57_d40 -268.48
bbob_f016_i58_d40 -1000.0
bbob_f016_i59_d40 -13.88
bbob_f016_i60_d40 -124.04
```

**Comment by nikohansen** [comment]
I could fix this (locally) in the Python interface, but there is some good chance that this is not only a problem in Python.

**Comment by nikohansen** [comment]
For the record, the reason why this is highly relevant: 
```python
>>> p = suite['bbob_f003_i01_d03']
>>> print(p([2,3,4e80]))
nan
>>> p = suite['bbob_f006_i01_d03']
>>> print(p([2,3,4e80]))
inf
```

**Comment by ttusar** [comment]
In C, the code sometimes breaks because of an assertion error, while on other occasions (for example in some of the transformations) the NAN is caught up by an else statement that sets the value to the optimal one (this must be why you get the optimum on the Rastrigin function). I'm not really sure why you don't get the assertion errors in Python...

In any case, what is the desired behavior in the C code for the case the decision vector has one or more NAN values? I would think that passing a NAN should return a NAN. @nikohansen, do you agree or have some other idea?

**Comment by nikohansen** [comment]
> In any case, what is the desired behavior in the C code for the case the decision vector has one or more NAN values? I would think that passing a NAN should return a NAN.

That looks perfectly right to me.

**Comment by nikohansen** [comment]
> the NAN is caught up by an else statement that sets the value to the optimal one

that I would consider as a serious bug, which definitely needs to be fixed.

**Comment by ttusar** [comment]
I agree and I'm on it.

**Comment by ttusar** [comment]
This is now fixed (awaiting to be pushed to the development branch, therefore I'm leaving the issue open until it is completed). 

Note this fixes only NAN-related problems. The overflow issue is still unsolved.

**Comment by ttusar** [comment]
Solved by a telling pull request number #888
